microros_extensions/zephyr_serial_transport.c

```diff
@@ -45,17 +68,17 @@ size_t uxr_write_serial_data_platform(uxrSerialPlatform* platform, uint8_t* buf,
 }
 
 size_t uxr_read_serial_data_platform(uxrSerialPlatform* platform, uint8_t* buf, size_t len, int timeout, uint8_t* errcode){ 
+    size_t read = 0;
+    int spent_time = 0;
 
-    size_t index = 0;
-
-    int ret = 0;
-    while (index < len && ret == 0)
-    {
-        ret = uart_poll_in(platform->uart_dev, &buf[index]);
-        if (ret == 0) {
-            index++;
-        }
+    while(ring_buf_is_empty(&in_ringbuf) && spent_time < timeout){
+        k_sleep(K_MSEC(1));
+        spent_time++;
     }
-    
-    return index;
+
+    uart_irq_rx_disable(platform->uart_dev);
+    read = ring_buf_get(&in_ringbuf, buf, len);
```

Comment:
Be careful, `read` is a posix function and I don't know if it exis in Zephyr but could be.

Reply:
Fixed